slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -888,6 +935,8 @@ void SLAMServiceImpl::ProcessDataAndStartSavingMaps(double data_start_time) {
             std::lock_guard<std::mutex> lk(viam_response_mutex);
             latest_global_pose = tmp_global_pose;
         }
+        if (action_mode == ActionMode::LOCALIZING) BackupLatestMap();
```

Comment:
I don't understand we we need to call BackupLatestMap here, since the purpose of BackupLatestMap is to save a cached map to serve during optimization, and at this point, we're done with optimization.

This also makes concurrency unsafe, since BackupLatestMap will modify latest_pointcloud_map, but now it's possible to access latest_pointcloud_map without holding optimization_lock, as long as localization_map_ready is true.

Reply:
`BackupLatestMap` holds the `viam_response_mutex`, so I figured that this would still be safe to do. Thinking about this again I probably do not need to run this for localization mode at all.

Reply:
Never mind, I missed that the viam_response_mutex is responsible for controlling access to latest_pointcloud_map, so I think this is safe. I still suggest not calling BackupLatestMap here, if we don't need it.